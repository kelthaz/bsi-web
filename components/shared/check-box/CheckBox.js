import React from 'react';
import PropTypes from 'prop-types';
import styles from './check-box.module.scss';

const CheckBox = ({ name, value, onChange, children, label }) => (
  <div className={`${styles['container-check-text']}`}>
    <div className={`${styles['container-input']}`}>
      <input
        name={name}
        className={` ${styles['my-check-box']}`}
        type="checkbox"
        value={label}
        onChange={onChange}
        checked={Array.isArray(value) ? value.indexOf(label) !== -1 : value}
      />
    </div>
    <div className={`${styles['container-content']}`}>{children}</div>
  </div>
);

CheckBox.defaultProps = {
  label: undefined,
};

CheckBox.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.any.isRequired,
  children: PropTypes.any.isRequired,
};

export default CheckBox;
