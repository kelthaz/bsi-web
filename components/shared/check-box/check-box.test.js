import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import React from 'react';
import CheckBox from './CheckBox';

describe('Pruebas en el componente CheckBox', () => {
  const onChange = jest.fn();
  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = {
      name: 'meEjercenControlMoralComoMoral',
      value: false,
      label: 'si',
      onChange,
    };
    // act
    const wrapper = shallow(
      <CheckBox {...props}>
        <p>Sí</p>
      </CheckBox>
    );
    // assert

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de llamarse onChange', () => {
    // arrange
    const valorEsperado = { target: { checked: true } };
    const props = {
      name: 'meEjercenControlMoralComoMoral',
      value: false,
      label: 'si',
      onChange,
    };
    // act
    const wrapper = shallow(
      <CheckBox {...props}>
        <p>Sí</p>
      </CheckBox>
    );
    wrapper.find('input').simulate('change', valorEsperado);
    // assert
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(valorEsperado);
  });

  test('Deben estar solo el primer check', () => {
    // arrange
    const propFirstCheck = {
      name: 'animales',
      value: ['Gatos'],
      label: 'Gatos',
      onChange,
    };
    const propSecondCheck = {
      name: 'animales',
      value: ['Gatos'],
      label: 'Perros',
      onChange,
    };
    // act
    const wrapper = mount(
      <div>
        <CheckBox {...propFirstCheck}>
          <p>Gatos</p>
        </CheckBox>
        <CheckBox {...propSecondCheck}>
          <p>Perros</p>
        </CheckBox>
      </div>
    );
    const { checked: firstChecked } = wrapper.find('input').at(0).props();
    const { checked: secondChecked } = wrapper.find('input').at(1).props();
    // assert
    expect(firstChecked).toBe(true);
    expect(secondChecked).toBe(false);
  });
});
