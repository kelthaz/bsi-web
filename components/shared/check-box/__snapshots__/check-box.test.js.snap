// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CheckBox Debe de mostrarse correctamente el componente 1`] = `
<div
  className="container-check-text"
>
  <div
    className="container-input"
  >
    <input
      checked={false}
      className=" my-check-box"
      name="meEjercenControlMoralComoMoral"
      onChange={[MockFunction]}
      type="checkbox"
      value="si"
    />
  </div>
  <div
    className="container-content"
  >
    <p>
      SÃ­
    </p>
  </div>
</div>
`;
